src/dmd/dinterpret.d

```diff
@@ -6836,6 +6870,70 @@ private Expression scrubCacheValue(Expression e)
     return e;
 }
 
+/********************************************
+ * Transitively replace all Expressions allocated in ctfeGlobals.region
+ * with Mem owned copies.
+ * Params:
+ *      e = possible ctfeGlobals.region owned expression
+ * Returns:
+ *      Mem owned expression
+ */
+private Expression copyRegionExp(Expression e)
+{
+    if (!e)
+        return e;
```

Comment:
instead of returning early and thereby confusing the branch predictor you could jump to a Label placed before the return statement in here.

Reply:
That's a bad idea in terms of source readability, and there is no telling how an optimizing compiler will lay out the branches anyway. If this was really important, `likely` intrinsics would be the way to go.